vector.go

```diff
@@ -32,22 +34,37 @@ type VectorField interface {
 const (
 	EuclideanDistance = "l2_norm"
 
-	// dotProduct(vecA, vecB) = vecA . vecB = |vecA| * |vecB| * cos(theta);
-	//  where, theta is the angle between vecA and vecB
-	// If vecA and vecB are normalized (unit magnitude), then
-	// vecA . vecB = cos(theta), which is the cosine similarity.
-	// Thus, we don't need a separate similarity type for cosine similarity
-	CosineSimilarity = "dot_product"
+	InnerProduct = "dot_product"
+
+	CosineSimilarity = "cosine"
 )
 
 const DefaultSimilarityMetric = EuclideanDistance
 
 // Supported similarity metrics for vector fields
 var SupportedSimilarityMetrics = map[string]struct{}{
 	EuclideanDistance: {},
+	InnerProduct:      {},
 	CosineSimilarity:  {},
 }
 
+func NormalizeVector(vector []float32) []float32 {
+	// first calculate the magnitude of the vector
+	var mag float32
+	for _, v := range vector {
+		mag += v * v
+	}
+	mag = float32(math.Sqrt(float64(mag)))
+	// zannot normalize a zero vector
```

Comment:
Nit: "zannot".

Reply:
Also may be worth adding a check to avoid unnecessarily normalising a unit vector(a vector with magnitude 1).

Reply:
done macha